static/js/components/RecentGcnEvents.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+
+import Paper from "@material-ui/core/Paper";
+import Typography from "@material-ui/core/Typography";
+import DragHandleIcon from "@material-ui/icons/DragHandle";
+import Button from "@material-ui/core/Button";
+import Chip from "@material-ui/core/Chip";
+import { makeStyles } from "@material-ui/core/styles";
+
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import relativeTime from "dayjs/plugin/relativeTime";
+
+import * as profileActions from "../ducks/profile";
+import * as recentEventsActions from "../ducks/recentGcnEvents";
+import WidgetPrefsDialog from "./WidgetPrefsDialog";
+
+dayjs.extend(relativeTime);
+dayjs.extend(utc);
+
+const useStyles = makeStyles((theme) => ({
+  header: {},
+  eventListContainer: {
+    height: "calc(100% - 5rem)",
+    overflowY: "auto",
+    marginTop: "0.625rem",
+    paddingTop: "0.625rem",
+  },
+  eventList: {
+    display: "block",
+    alignItems: "center",
+    listStyleType: "none",
+    paddingLeft: 0,
+    marginTop: 0,
+  },
+  eventNameContainer: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  eventNameLink: {
+    color: theme.palette.primary.main,
+  },
+  eventTags: {
+    marginLeft: "1rem",
+    "& > div": {
+      margin: "0.25rem",
+      color: "white",
+      background: theme.palette.primary.main,
+    },
+  },
+  BNS: {
+    background: "#468847!important",
+  },
+  NSBH: {
+    background: "#b94a48!important",
+  },
+  BBH: {
+    background: "#333333!important",
+  },
+  GRB: {
+    background: "#f89406!important",
+  },
+  AMON: {
+    background: "#3a87ad!important",
+  },
+  Terrestrial: {
+    background: "#999999!important",
+  },
+}));
+
+const defaultPrefs = {
+  maxNumEvents: "",
+  sinceDaysAgo: "",
+};
+
+const RecentGcnEvents = ({ classes }) => {
+  const styles = useStyles();
+
+  const { gcnEvents } = useSelector((state) => state.recentGcnEvents);
```

Comment:
```suggestion
  const gcnEvents = useSelector((state) => state.gcnEvents);
```
to accord with comments below